kind, sriov: Introduce kind sriov bump, generic provider name, and bump to k8s-1.28

oshoval opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. Consider creating this PR as draft: https://github.com/kubevirt/kubevirt/blob/main/CONTRIBUTING.md#consider-opening-your-pull-request-as-draft
2. Follow the instructions for writing a release note from k8s: https://git.k8s.io/community/contributors/guide/release-notes.md
-->

**What this PR does / why we need it**:
Introduce bump-kind-sriov script
The script eases bumping kind-sriov provider.
Later we can use it for automation.

Auto detect kubectl path
Once in a while the kubectl path in kind node
is changed (for example if the base image is changed).
So far it was only one of two paths, and we are using it
only for copying kubectl from the node.
Auto detect it, so we won't need to change it whenever it changes.

Duplicate kind-1.27-sriov, as kind-sriov.
Using kind-sriov generic name, it will allow easier updating
of the CI jobs once we update the provider.

Extract kind-sriov kind release and node image parameters to files
It allows updating them in more elegant way,
without changing source files themselves.

Bump kind-sriov to v0.19.0, k8s-1.28

Fix bash error "unary operator expected".

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
Will create the required jobs, tested it on CI, but best to test also the final version.
Later we can adapt vgpu and kind-1.28 to also support the bump script.
Need to fix README.
v0.20.0 bump should be done on a follow-up, as it doesn't work out of the box on CI.

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note
None
```

oshoval commented:
/test check-up-kind-1.27-sriov

oshoval commented:
/test check-up-kind-1.27-sriov

oshoval commented:
/test check-up-kind-1.27-sriov

oshoval commented:
/test check-up-kind-1.27-sriov

oshoval commented:
/test check-up-kind-1.27-sriov
/test check-up-kind-1.27-vgpu

EdDev commented on cluster-up/cluster/kind-sriov/provider.sh:
```diff
@@ -21,8 +21,11 @@ function print_available_nics() {
 function set_kind_params() {
-    export KIND_VERSION="${KIND_VERSION:-0.18.0}"
-    export KIND_NODE_IMAGE="${KIND_NODE_IMAGE:-kindest/node:v1.27.1@sha256:9915f5629ef4d29f35b478e819249e89cfaffcbfeebda4324e5c01d53d937b09}"
+    version=$(cat cluster-up/cluster/$KUBEVIRT_PROVIDER/version)
```
nit: Using `read` to read the first line would be nicer, but this is not that important.

EdDev reviewed (commented):
Very nice, thank you!

The duplication of the old kind is a bit risky (and also harder to review, as it require trust that you copied it without modification).
I just hope you keep this intermediate state where both are there limited so they will not diverge.

Note, you could have used a soft-link and kept the version as is without a bump. Then in a separate PR to bump it, showing how minimal the change is.

oshoval commented:
Thanks
It is so it will allow smoother update, otherwise when merging this PR, until the project-infra one that
renames the job is presented, the jobs would fail.
I will make sure that when i delete the old one it doesnt diverges beside the known changes.

oshoval commented:
Moved the bump script under hack/kind
and added a provision manage rule that will skip this folder for vm based provider reprovision
because it doesnt affect them, and if it was on `hack/` it does affect them

oshoval commented:
Not related to this PR but conformance are failing on all 3 vm based providers
Hi @brianmcarey, fyi

oshoval commented:
/retest
lets see if it was one timer (but it is weird that they all 3 failed still)

oshoval commented:
seems there is a regression, maybe some rpm change or kernel update, as nothing was changed beside
(according that old provider did pass, and this one didnt 2/2)

oshoval commented:
Change the folder where the script exists, so we wont need to provision nothing in this PR.
As we can see reprovision has a bug that should be fixed but shouldn't block this PR.
This one of the reasons provision manager was meant for.
